### PR TITLE
webhook

### LINKED ISSUE #343: add webhook url to clients
- [ ] add a field `webhookUrl` to clients.
- [ ] default value should be NULL
- [ ] validate it is a valid url when it is added to a client

### PR DESCRIPTION
**Pull request by VibhorCodecianGupta**
fixes #343 and #345

### CODE REVIEW

File: src/db/models.js [hunk 1]
```diff
@@ -94,6 +94,7 @@ const Client = db.define('client', {
     secret: Sequelize.DataTypes.STRING,
     domain: Sequelize.DataTypes.ARRAY(Sequelize.DataTypes.STRING),
     callbackURL: Sequelize.DataTypes.ARRAY(Sequelize.DataTypes.STRING),
+    webhookURL: {type: Sequelize.DataTypes.STRING, default: ''},
```
Comment (championswimmer) [inline src/db/models.js]: default NULL

File: src/routers/api/clients.js [hunk 2]
```diff
@@ -22,6 +22,11 @@ router.post('/add', async function (req, res) {
         clientCallbacks : req.body.callback.replace(/ /g, '').split(';'),
         defaultURL : req.body.defaulturl.replace(/ /g, '')
     }
+    const regex = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/
```
Comment (championswimmer) [inline src/routers/api/clients.js]: move the isurl check into utils, it'll be helpful other places too I guess
https://github.com/coding-blocks/oneauth/blob/master/src/utils/urlutils.js

**Comment by VibhorCodecianGupta** [comment]
@championswimmer done

File: src/routers/api/clients.js [hunk 3]
```diff
@@ -22,6 +22,10 @@ router.post('/add', async function (req, res) {
         clientCallbacks : req.body.callback.replace(/ /g, '').split(';'),
         defaultURL : req.body.defaulturl.replace(/ /g, '')
     }
+    
+    if (req.body.webhookURL && isURL(req.body.webhookURL)){
+        options.webhookURL = req.body.webhookURL
```
Comment (championswimmer) [inline src/routers/api/clients.js]: if not, then you should send an error, or how else will the user know what went wrong in adding the webhook ?

File: views/client/add.hbs [hunk 4]
```diff
@@ -24,6 +24,10 @@
                     <div class="label-input col-md-4 col-sm-3">Default Url</div>
                     <input type="text" class="input-text col-8" placeholder="Default URL" name="defaulturl"/>
                 </div>
+                <div class="row no-gutters align-items-center justify-content-between detail-row">
+                    <div class="label-input col-md-4 col-sm-3">Webhook Url</div>
+                    <input type="text" class="input-text col-8" placeholder="Webhook URL" name="webhookURL"/>
```
Comment (championswimmer) [inline views/client/add.hbs]: input type="url" not text

File: src/routers/api/clients.js (hunk 3 above)
Comment (VibhorCodecianGupta) [reply src/routers/api/clients.js]: Done @championswimmer

**Comment by VibhorCodecianGupta** [comment]
\

**Pull request closed without merging**